instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/AwsLambdaFunctionInstrumenter.java

```diff
@@ -46,15 +47,25 @@ public void end(
   }
 
   public Context extract(AwsLambdaRequest input) {
-    return ParentContextExtractor.extract(input.getHeaders(), this);
-  }
-
-  public Context extract(Map<String, String> headers, TextMapGetter<Map<String, String>> getter) {
     ContextPropagationDebug.debugContextLeakIfEnabled();
 
     return openTelemetry
         .getPropagators()
         .getTextMapPropagator()
-        .extract(Context.root(), headers, getter);
+        .extract(Context.root(), input.getHeaders(), MapGetter.INSTANCE);
+  }
+
+  private enum MapGetter implements TextMapGetter<Map<String, String>> {
+    INSTANCE;
+
+    @Override
+    public Iterable<String> keys(Map<String, String> map) {
+      return map.keySet();
+    }
+
+    @Override
+    public String get(Map<String, String> map, String s) {
+      return map.get(s.toLowerCase(Locale.ROOT));
```

Comment:
why do you need to use `toLowerCase` here?

Reply:
Because the headers map is converted to lower case https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/09577f97ffc0ca9f703386e12afbd018d12a2935/instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/MapUtils.java#L18